src/contexts/walletconnect.tsx

```diff
@@ -52,43 +55,27 @@ import {
   TxDataToModalType,
   TxType
 } from '@/types/transactions'
+import { SessionProposalEvent, SessionRequestEvent } from '@/types/walletConnect'
 import { AlephiumWindow } from '@/types/window'
 import { WALLETCONNECT_ERRORS } from '@/utils/constants'
-
-type RequestEvent = SignClientTypes.EventArguments['session_request']
-type ProposalEvent = SignClientTypes.EventArguments['session_proposal']
-
-type WalletConnectSessionState = 'uninitialized' | 'proposal' | 'initialized'
+import { getActiveWalletConnectSessions, isNetworkValid, parseSessionProposalEvent } from '@/utils/walletConnect'
 
 export interface WalletConnectContextProps {
   walletConnectClient?: SignClient
-  requestEvent?: RequestEvent
-  proposalEvent?: ProposalEvent
+  pairWithDapp: (uri: string) => void
+  unpairFromDapp: (pairingTopic: string) => Promise<void>
   dappTxData?: DappTxData
-  onSessionRequestError: (event: RequestEvent, error: ReturnType<typeof getSdkError>) => Promise<void>
-  onSessionRequestSuccess: (event: RequestEvent, result: node.SignResult) => Promise<void>
-  onSessionDelete: () => void
-  connectToWalletConnect: (uri: string) => void
-  requiredChainInfo?: ChainInfo
-  wcSessionState: WalletConnectSessionState
-  sessionTopic?: string
-  onProposalApprove: (topic: string) => void
-  connectedDAppMetadata?: ProposalEvent['params']['proposer']['metadata']
```

Comment:
As seen here, the WC context is now much more minimal than before, similar to the mobile wallet.

Reply:
Much better!

Reply:
Incredible work on this file. It must have required a ton of patience 👏